terminal_colors/terminal_colors.py

```diff
@@ -554,10 +560,22 @@ def _terminal():
         terminal class
     """
     curses.setupterm()
-    num_colors = curses.tigetnum('colors')
-    return {16:term16, 88:term88, 256:term256}.get(num_colors, term16)
+    if options.force_term is None:
+        num_colors = curses.tigetnum('colors')
+    else:
+        num_colors = options.force_term
+    if num_colors > 0:
+        return TERM_TABLE.get(num_colors, term16)
+
 
+TERM_TABLE = {16:term16, 88:term88, 256:term256}
 def main():
+    global options
+    # instantiate global options based on command arguments
+    options = _get_options(sys.argv[1:])
+    # don't allow -f by itself
+    options.foreground = options.foreground and (
+            options.numbers or options.hex or options.rgb)
```

Comment:
I moved the execution of the Option Parser out of the auto-execute section (AES) of the script because `_get_options` now references `TERM_TABLE`, which is declared late in the AES. I suppose I could have kept it out of `main()` and just put it below the declaration of `TERM_TABLE` in the AES. But anyway, this should work fine.